.github/workflows/jekyll.yml

```diff
@@ -1,20 +1,76 @@
 name: Jekyll site CI
-
 on:
-  push:
-    branches: [ changes, release, translate ]
   pull_request:
-    branches: [ changes, release, translate ]
-
+    branches: [ next-release, release ]
+  push:
+    branches: [ next-release ]
 jobs:
-  build:
+  jekyll_site_ci:
     permissions:
-      contents: read
+      contents: write
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - name: Build the site in the jekyll/builder container
-      run: |
-        docker run \
-        -v ${{ github.workspace }}:/srv/jekyll -v ${{ github.workspace }}/_site:/srv/jekyll/_site \
-        jekyll/builder:latest /bin/bash -c "chmod a+w /srv/jekyll/Gemfile.lock && chmod 777 /srv/jekyll && jekyll build --future"
+      - uses: actions/checkout@v2
+
+# Retrieve po4a installation cache or create it if not found:
+      - name: Cache po4a dependencies
+        uses: actions/cache@v1.0.3
+        id: cache-po4a
+        with:
+          path: "~/po4a"
+          key: ${{ runner.os }}-po4a
+      - name: Install or retrieve po4a from cache
+        env:
+          CACHE_HIT: ${{steps.cache-po4a.outputs.cache-hit}}
+        run: |
+          if [[ "$CACHE_HIT" == 'true' ]]; then
+            sudo cp --force --recursive ~/po4a/* /
+          else
+            sudo apt install -yq gettext libsgmls-perl libyaml-tiny-perl opensp
+            wget -O po4a.deb https://github.com/jamulussoftware/assets/raw/main/po4a/po4a_0.64-alpha.deb
+            sudo dpkg -i po4a.deb
+            mkdir -p ~/po4a
+            for dep in po4a libcroco3 libosp5 sgml-base gettext libsgmls-perl libyaml-tiny-perl opensp; do
+                dpkg -L $dep | while IFS= read -r f; do if test -f $f; then echo $f; fi; done | xargs cp --parents --target-directory ~/po4a/
+            done
+          fi
+
+# Paths changes filter. If there's been a change to any files defined in the filter, the step (update templates) runs:
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          base: ${{ github.ref }}
+          filters: |
+            templates:
+              - 'wiki/en/*.md'
+      - name: Update templates
+        if: ${{ steps.filter.outputs.templates == 'true' || github.event.pull_request.title == 'Add new language' }}
```

Comment:
You shouldn't only run this if a PR is named "Add new language". Everybody could trigger it (not only people with write/maintenance access), and what happens if someone labels his PR  "Add chinese translation"?

I think it should be more generic: Maybe check for a specific PR label ("new language") or similar.

Reply:
I'm not sure I understand. Do you want to make it easier to trigger this or more restrictive? As it is, anyone can trigger it by creating the folder + empty file and the PR title "Add new language". It's documented in the README in _translator-files/ so people know exactly how to do it. As commented elsewhere, this is going to be a pretty rare event, and anyone wanting to do it just has to be directed to the relevant README. If you have to instruct them to add a specific label instead of a PR title, you're not really simplifying it, are you? It's just a different instruction.

That said, I have no problem changing how this is done - I simply need to understand better what the goal is.

Reply:
Goal: no need to have a fixed PR title
Solution: don’t check for a title, but for a fixed label (which e.g. maintainers would add).
Open question: do we have security issues/bad behaviour if a PR author (=untrustworthy) opens a PR which triggers adding a new language

I hope it’s clearer now

Reply:
Ok - so you'd like to make it more restrictive, i.e. it gets triggered when a maintainer adds a label? Yes, I agree that there's the risk a cheeky person could open a PR, but... that's the case at present, for any public repo. The new language files get added to their PR, they're not automatically pushed anywhere, so if it's an unwanted/untrusted PR you just have to reject it.

Reply:
Ok, so adding a new language triggered by adding a label (e.g. "new language") works too, limiting the scope of who can trigger the action. Either way though, I think the approach is flawed if the PR author has to enable actions for their repo (since adding a language pushes new files to their repo)... But only triggering it on push won't work either because empty folders containing empty files don't get pushed from one branch to another (while they can be pushed from a local branch), so another mechanism would have to be devised...

 Also, as an aside, zipping the site every time jekyll.yml is run means we'd be ending up with a whole load of zipped files being uploaded. Adding a language would cause 3 uploads (PR, add label, merge). 

It's back to the drawing board I think - suggestions are welcome!

Reply:
> Either way though, I think the approach is flawed if the PR author has to enable actions for their repo (since adding a language pushes new files to their repo)... But only triggering it on push won't work either because empty folders containing empty files don't get pushed from one branch to another (while they can be pushed from a local branch), so another mechanism would have to be devised...

Good points. Didn't think about the issue of needing to enable actions. Actions do also run on our side and can push to the repo of the new user, however this might not be wanted.

What about checking for a git tag (but this will then only work for the main repo)?

new-lang-[language name] --> Creates folders and files --> pushes them to the repo --> now a person forks the repo --> translation work --> open PR --> merge new language?

Reply:
Hmm - I was hoping for something -a couple of lines- that could be added to the workflow to automate adding a new language because at the end of the day, doing it manually is so trivial that I think coming up with convoluted workarounds to do something that will seldom be needed is a bit of a waste of everyone's time... In the time I've taken just to think about how to do this I could have added 200 more languages =)

Another idea: dorny/paths-filter@v2 has an interesting filter that can be triggered when you add/modify/delete files in a specified path. I set it to trigger update-templates if files are added to _translator-files/po/ and it works nicely. It can be set to do this on:push so it only creates the .po files on our repo when the PR is merged and doesn't require activating actions by the PR author. So:

- User creates empty lang folder + temp file and makes PR. We could also do this for them, but whatever.
- When PR is merged, the language files are created and pushed to the main repo.
- User pulls update/forks from main repo and translates.

The good thing is it's only takes 3 lines added to the script.

Reply:
Hmm. Yes, I think your suggestion is the easiest. Let’s just do that.

Reply:
Done. The procedure is as explained above. No special PR titles or anything - the .po files get created and added to the main repo when merging a PR submitted with a new language folder containing a dummy temp.po~ file.

Reply:
BTW, I intentionally didn't include the "add new language" function in main.yml as I think languages should only be added to 'next-release'.

Reply:
> What about checking for a git tag (but this will then only work for the main repo)?
> 
> new-lang-[language name] --> Creates folders and files --> pushes them to the repo --> now a person forks the repo --> translation work --> open PR --> merge new language?

I had a go at this:
- User creates an issue with the new language in square brackets in the title (e.g. "Add Basque [eu]")
- Adding a "new language" label triggers a workflow that extracts the language from the square brackets, creates the folder/files and pushes them to the 'next-release' branch (not to 'release').
- Ready for the user to fork and translate the new language.

I think it's basically what you were proposing - I thought it would be harder than it actually was. I did do it in a separate workflow script though, and I omitted the site zipping and uploading, but left the building part as a check.

The only thing that doesn't seem to work is retrieving the cached dependencies - it says "issues" event types are not supported. It still runs fine though, just takes a little longer.

So there's that option too. It is neater than what I have now.

Reply:
Yes, this sounds good. You just need to make sure that the string in brackets doesn’t open a security hole (you need to validate it).

Reply:
Uh... ok (just looked up what that means). So should I check if it matches a two-letter string or something like that?

Reply:
Exactly. 
en_US,en, should be allowed, but not ../

You can probably just modify the regular expression in the sed command.

Reply:
Done. Restricted to 'xx' and 'xx_XX' formats - anything else will cause it to exit.

Reply:
Added the script to this PR.